Update msiv1_token_revocation.md (#5222)
AzureAD/microsoft-authentication-library-for-dotnet: docs/msiv1_token_revocation.md

Steps 5-9 are new and show how the RP propagates the revocation signal.

### Explanation:

> 1. CX creates an **MSAL Client** with `.WithClientCapabilities(cp1)`, to let the token issuer that it is capable of handling token revocations.

Reviewer: ```suggestion
1. CX creates an **MSAL Client** with `.WithClientCapabilities(cp1)`, to let the token issuer know that it is capable of handling token revocations.
```

> 1. CX creates an **MSAL** **Client** with `.WithClientCapabilities(cp1)`, to let the token issuer that it is capable of handling token revocations.

Reviewer 2: nit:
```suggestion
1. CX creates an **MSAL Client** with `.WithClientCapabilities(cp1)`, to let the token issuer that it is capable of handling token revocations.
```

Author: Updated

2. The client (CX) calls some **Resource** with token **T**.
3. The resource detects **T** is bad (revoked) and returns **401** + **claims C**.
4. CX parses the WWW-Authenticate header, extracts the claims **C** and uses MSAL **AcquireToken** with `.WithClaims(C)`.
5. MSAL inspects its cache first. If it finds a token, the token is considered to have been revoked. MSAL needs to tell the token issuer about it, so that the token issuer can also bypass its own cache.
6. MSAL calls **MITS** with `xms_cc=cp1&token_sha256_to_refresh=SHA256(T)`.

> 7. The token issuer uses the information to bypass its own caches and to get a new token from eSTS.

Reviewer: ```suggestion
7. **MITS** uses the information to bypass its own caches and to get a new token from its upstream **SFRP**.
```

8. **SFRP** uses MSAL again to get a **new** token from eSTS.

> [!IMPORTANT]
> This design is only applicable to MIRP api-version=2025-03-30 (for App Service). api-version for service fabric will be soon made available. 

> [!NOTE]

> > The SHA256 conversion is done by doing a Base64-encoded SHA-256 hash of the token (UTF-8). For example: Convert.ToBase64String(SHA256(Encoding.UTF8.GetBytes(accessToken))). 

Reviewer: Also need to change the `Convert.ToBase64String(...)` part.

> > The SHA256 conversion is done by doing a Base64-encoded SHA-256 hash of the token (UTF-8). For example: Convert.ToBase64String(SHA256(Encoding.UTF8.GetBytes(accessToken))). 

Author: Updated

> > Example - "test_token" -> "cc0af97287543b65da2c7e1476426021826cab166f1e063ed012b855ff819656"

Reviewer 2: Can we check this in the repo? Without code ql prompting us to remove the secrets?

Author: I am not sure how we can do that 
For MSAL Go it did not trigger as it does not look like token, I guess. https://github.com/AzureAD/microsoft-authentication-library-for-go/pull/567

Reviewer 2: Sounds good. Thanks!

### Explanation:
1. The client (CX) calls some **Resource** with token **T**.
2. The resource detects **T** is bad (revoked) and returns **401** + **claims C**.

> 3. CX parses **C** and calls **MSAL** **Client** with `.WithClientCapabilities(cp1)`.

Reviewer 3: ```suggestion
3. CX parses the WWW-Authenticate header, extracts the claims **C** and calls **MSAL** **Client** with `.WithClientCapabilities(cp1)`.
```

> 4. MSAL calls **AcquireToken** with `.WithClaims(C)`.

Reviewer 3: ```suggestion
4. CX parses the WWW-Authenticate header, extracts the claims **C** and uses MSAL **AcquireToken** with `.WithClaims(C)`.
```

> 5. MSAL sees the local cached token is "bad" → triggers a refresh flow.

Reviewer 3: ```suggestion
5. MSAL inspects its cache first. If it finds a token, the token is considered to have been revoked. MSAL needs to tell the token issuer about it, so that the token issuer can also ignore it.
```

6. MSAL calls **MITS** with `xms_cc=cp1&token_sha256_to_refresh=SHA256(T)`.

> 7. **MITS** is basically a proxy, forwarding the query to **SFRP**.

Reviewer 3: ```suggestion
7. The token issuer uses the information to bypass its own caches and to get a new token from eSTS.
```

8. **SFRP** uses MSAL again to get a **new** token from eSTS.

> [!IMPORTANT]
> This design is only applicable to MIRP api-version=2025-03-30 (for App Service). api-version for service fabric will be soon made available. 

> [!NOTE]

> > The `token_sha256_to_refresh=SHA256(T)` here the SHA256 converts the token into a SHA256 string. Example - "examplestring" -> output

Reviewer 3: ```suggestion
> The SHA256 conversion is done by doing a Base64-encoded SHA-256 hash of the token (UTF-8). For example: Convert.ToBase64String(SHA256(Encoding.UTF8.GetBytes(accessToken))). 
> Example - "test_token" -> "zAr5codUO2XaLH4UdkJgIYJsqxZvHgY+0BK4Vf+BllY="

```

Reviewer 3: https://dotnetfiddle.net/xMmbvP
